include/rmm/mr/device/tracking_resource_adaptor.hpp

```diff
@@ -133,27 +177,153 @@ class tracking_resource_adaptor final : public device_memory_resource {
    * @return std::size_t number of bytes that have been allocated through this
    * allocator.
    */
-  std::size_t get_allocated_bytes() const noexcept { return allocated_bytes_; }
+  std::size_t get_allocated_bytes() const noexcept
+  {
+    return get_total_allocation_counts().current_bytes;
+  }
 
   /**
-   * @brief Log any outstanding allocations via RMM_LOG_DEBUG
+   * @brief Returns an allocation_counts struct for this adaptor containing the
+   * total current, peak, and total number of bytes and allocation count for
+   * this adaptor regardless of any push/popped allocation_counts. Note: Because
+   * its possible to change memory resources at any time while maintaining the
+   * same upstream memory resource, its possible to have a negative allocation
```

Comment:
I don't understand the note, or the reason for a stack. You can't change the upstream for a `resource_adaptor`, you would have to create a new resource adaptor. You can't have more deallocations than allocations without an exception.  This seems way overcomplicated.